userspace/libsinsp/container_engine/bpm.cpp

```diff
@@ -61,6 +61,7 @@ bool bpm::resolve(sinsp_threadinfo *tinfo, bool query_os_for_missing_info)
 	if(container_cache().should_lookup(container_info.m_id, CT_BPM))
 	{
 		container_info.m_name = container_info.m_id;
+		container_info.set_lookup_status(sinsp_container_lookup::state::SUCCESSFUL);
```

Comment:
I was wondering whether it would be better to set the SUCCESSFUL lookup status inside
`container_cache().notify_new_container()`: it requires less changes and it "feels" better, ie: since we are notifying the new container, we mark the container as successfully looked up.

Reply:
It feels less error prone too (like: a new engine gets introduced and we forgot to add the `set_lookup_state` call).

Reply:
Except some container engines (e.g. `libvirt_lxc`) add the container to the cache and already expect the lookup status to be SUCCESSFUL, so that notify_new_container() doesn't do anything.

As stated more at the beginning of the PR this change is just an initial improvement so that the lookup state is not SUCCESSFUL by default, which makes not a lot of sense and can backfire even more. We have this issue (https://github.com/falcosecurity/libs/issues/1708) we wanted to work on after Falco 0.38.0.

Reply:
Oh sorry, i completely overlooked that. Then we are good to go!